src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -454,7 +454,18 @@ public override bool CanVerifySignedPackages(SignedPackageVerifierSettings verif
             }
             else if (RuntimeEnvironmentHelper.IsLinux || RuntimeEnvironmentHelper.IsMacOSX)
             {
-                return false;
+                // Conditionally enable back package sign verification temporary disabled due to Mozilla drop Symantec as CA on Linux/MAC.
+                // Please note: Linux/MAC case sensitive for env var.
+                string signVerifyEnvVariable = Environment.GetEnvironmentVariable("DOTNET_OPT_IN_SECURE_PACKAGE_VERIFICATION");
```

Comment:
Use the https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/EnvironmentVariableWrapper.cs so you can add a unit test. 

This is an API that's easily unit testable.

Reply:
I already checked that one. I don't see any actual benefit.
Does it prevent from cross test `env var` value set contamination? If yes then we're going to have more flaky unit tests?
Specially it's problem if parallel tests are running same time.

Reply:
Yes it prevents parallel unit test issues as you can provide a mock to each test with values you control. For example:

https://github.com/NuGet/NuGet.Client/blob/f24bad0668193ce21a1db8cabd1ce95ba509c7f0/test/NuGet.Core.Tests/NuGet.Configuration.Test/ProxyCacheTests.cs#L96-L107

Reply:
> Does it prevent from cross test env var value set contamination? If yes then we're going to have more flaky unit tests?

Yes, because as @loic-sharma showed, you can mock the specific instance of the wrapper.

Reply:
I just realized problem with approach is I have to make change to public[ `PackageExtractor.ExtractPackageAsync`](https://github.com/NuGet/NuGet.Client/blob/7e3b0e445edecbdb6c46895ee4375cf834f45d87/src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs#L20) since it doesn't accept IEnvironmentVariableReader as parameter. Is it really justifying change to public api just sake of unit test?

Reply:
I just realized problem with this approach. I have to make change to existing public api [PackageExtractor.ExtractPackageAsync](https://github.com/NuGet/NuGet.Client/blob/7e3b0e445edecbdb6c46895ee4375cf834f45d87/src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs#L20) in order to pass `IEnvironmentVariableReader` as parameter. 
Is it really good justification to make change to public api, just sake of unit testing working?

Reply:
You can add an internal constructor.
We set internalsvisibleto to test projects. https://github.com/NuGet/NuGet.Client/blob/7e3b0e445edecbdb6c46895ee4375cf834f45d87/src/NuGet.Core/NuGet.Packaging/Properties/AssemblyInfo.cs#L11

Reply:
@nkolev92 
`PackageExtractor` is static class, so it doesn't have constructor. 
If we introduce new field into static class then it'll cause problem in concurrent unit test context.

Reply:
> PackageExtractor is static class, so it doesn't have constructor.


The change is in PackageArchiveReader not PackageExtractor.

Reply:
Ok. I create new internal constructors for `PackageArchiveReader` and created new tests in most places by creating duplicate of existing test with environmental variable change.

But there're 2 places I can't use it and I don't think it it's really important for now.
1. RestoreCommandTests.cs >> [RestoreCommand_InvalidSignedPackageAsync_SuccessAsync()](https://github.com/NuGet/NuGet.Client/blob/7e3b0e445edecbdb6c46895ee4375cf834f45d87/test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs#L1048)
2. PackageExtractorTests.cs>>[InstallFromSourceAsyncByStream_InvalidSignPackageWithoutUnzipAsync](https://github.com/NuGet/NuGet.Client/blob/7e3b0e445edecbdb6c46895ee4375cf834f45d87/test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs#L3099)

Please review again.